docling_core/types/experimental/document.py

```diff
@@ -0,0 +1,1168 @@
+    # Could stem from filesystem, source URI, Content-Disposition header, ...
+
+    uri: Optional[AnyUrl] = (
+        None  # any possible reference to a source file,
+        # from any file handler protocol (e.g. https://, file://, s3://)
+    )
+
+    @field_validator("binary_hash", mode="before")
+    @classmethod
+    def parse_hex_string(cls, value):
+        """parse_hex_string."""
+        if isinstance(value, str):
+            try:
+                # Convert hex string to an integer
+                hash_int = Uint64(value, 16)
+                # Mask to fit within 64 bits (unsigned)
+                return hash_int & 0xFFFFFFFFFFFFFFFF
+            except ValueError:
+                raise ValueError(f"Invalid sha256 hexdigest: {value}")
+        return value  # If already an int, return it as is.
+
+    @field_validator("mimetype")
+    @classmethod
+    def validate_mimetype(cls, v):
+        """validate_mimetype."""
+        # Check if the provided MIME type is valid using mimetypes module
+        if v not in mimetypes.types_map.values():
+            raise ValueError(f"'{v}' is not a valid MIME type")
+        return v
+
+
+class RefItem(BaseModel):
+    """RefItem."""
+
+    cref: str = Field(alias="$ref")
+
+    # This method makes RefItem compatible with DocItem
+    def get_ref(self):
+        """get_ref."""
+        return self
+
+    model_config = ConfigDict(
+        populate_by_name=True,
+    )
+
+    def resolve(self, doc: "DoclingDocument"):
+        """resolve."""
+        path_components = self.cref.split("/")
+        if len(path_components) > 2:
+            _, path, index_str = path_components
+        else:
+            _, path = path_components
+            index_str = None
+
+        if index_str:
+            index = int(index_str)
+            obj = doc.__getattribute__(path)[index]
+        else:
+            obj = doc.__getattribute__(path)
+
+        return obj
+
+
+class ImageRef(BaseModel):
+    """ImageRef."""
+
+    mimetype: str
+    dpi: int
+    size: Size
+    uri: AnyUrl
+
+    @field_validator("mimetype")
+    @classmethod
+    def validate_mimetype(cls, v):
+        """validate_mimetype."""
+        # Check if the provided MIME type is valid using mimetypes module
+        if v not in mimetypes.types_map.values():
+            raise ValueError(f"'{v}' is not a valid MIME type")
+        return v
+
+
+class ProvenanceItem(BaseModel):
+    """ProvenanceItem."""
+
+    page_no: int
+    bbox: BoundingBox
+    charspan: Tuple[int, int]
+
+
+class NodeItem(BaseModel):
+    """NodeItem."""
+
+    self_ref: str  # format spec: json-path
+    parent: Optional[RefItem] = None
+    children: List[RefItem] = []
+
+    model_config = ConfigDict(extra="forbid")
+
+    def get_ref(self):
+        """get_ref."""
+        return RefItem(cref=self.self_ref)
+
+
+class GroupItem(NodeItem):  # Container type, can't be a leaf node
+    """GroupItem."""
+
```

Comment:
Add provenance field to GroupItem or NodeItem,
to support provenance of entire lists, or slides (that are collections of elements)
In case of lists - it might be that individual bounding boxes are not possible or hard to obtain, but bounding box of an entire list is readily available. And in any way we would like to keep bounding boxes of lists, for this Group should support provenance.